### PR TITLE
fix: fix the logic of data update time using meta.json

### LINKED ISSUE #99: [Bug] Data is not the newest
### Current Behavior

I updated the leaderboard for 2023.6 and the [`meta.json`](https://xlab-open-source.oss-cn-beijing.aliyuncs.com/open_leaderboard/meta.json) file contains the correct timestamp, but the leaderboard still only shows the data for 2023.5 as the newest month.

### Expected Behavior

If the `meta.json` file gives a timestamp in 2023.7, then it means the data for 2023.6 has been updated, so it should be the newest data. Which means all the behavior of the leaderboard should depend on the `meta.json` file.

### Environment

```markdown
- OS: MacOS
- Browser: Chrome
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by andyhuang18**
## Brief Information

This pull request is in the type of ([more info](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#type) about types):

- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [x] fix
- [ ] perf
- [ ] refactor
- [ ] test

Related issues ([all available keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

- resolve #99 

## Details

- The previously set fixed time update data is very rigid, and currently the update can be controlled through `meta.json` data.
- Well...I hope the update speed at the beginning of each month can be faster...

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/X-lab2017/open-leaderboard/blob/main/README.md) doc

### CODE REVIEW

**Review by tyn1998 — COMMENTED** [review]
Hi @andyhuang18, in the changed files I didn't see any changes related to `meta.json`. Has the PR been ready or is it still in progress?

**Comment by andyhuang18** [comment]
> Hi @andyhuang18, in the changed files I didn't see any changes related to `meta.json`. Has the PR been ready or is it still in progress?

The original code logic will use `meta.json`, this PR deletes the previously set fixed-time update logic.

File: src/index.js [hunk 1]
```diff
@@ -14,26 +14,10 @@ const { Content } = Layout;
...
   'https://xlab-open-source.oss-cn-beijing.aliyuncs.com/open_leaderboard/meta.json';
 
-const getLastMonth = (lastUpdateTime) => {
-  let year = lastUpdateTime.getFullYear();
-  let monthIndex = lastUpdateTime.getMonth();
-  let date = lastUpdateTime.getDate();
-
-  if (date < 20) {
-    monthIndex--;
-  }
```
Comment (tyn1998) [inline src/index.js:26]: Now the last month = the month before the lastUpdateTime's month, `date < 20` is no longer needed.

File: src/index.js [hunk 2]
```diff
@@ -50,7 +34,9 @@ const App = () => {
   let year = null,
     monthIndex = null;
   if (lastUpdateTime) {
-    [year, monthIndex] = getLastMonth(lastUpdateTime);
+    const lastDataAvailableMonth = new Date(lastUpdateTime.setDate(0));
```
Comment (tyn1998) [inline src/index.js:37]: Date.prototype.setDate(0) is a better way to decrease month by 1.

**Review by tyn1998 — APPROVED** [review]

**Pull request merged**